database/schema/sqlite-schema.sql

```diff
@@ -0,0 +1,301 @@
+INSERT INTO migrations VALUES(68,'2023_12_08_000000_update_useraccounts_table',1);
+INSERT INTO migrations VALUES(69,'2023_12_16_000000_update_gamehashlibrary_table',1);
+INSERT INTO migrations VALUES(70,'2024_01_24_000000_update_codenotes_table',1);
+INSERT INTO migrations VALUES(71,'2024_02_02_000001_update_forum_topic_comment_table',1);
+INSERT INTO migrations VALUES(72,'2024_03_09_000000_update_gamehashlibrary_table',1);
+INSERT INTO migrations VALUES(73,'2024_03_16_000000_update_codenotes_table',1);
+INSERT INTO migrations VALUES(74,'2024_03_16_000001_update_comment_table',1);
+INSERT INTO migrations VALUES(75,'2024_03_16_000001_update_ticket_table',1);
+INSERT INTO migrations VALUES(76,'2024_03_17_000000_update_siteawards_table',1);
+INSERT INTO migrations VALUES(77,'2024_03_17_000000_update_subscription_table',1);
+INSERT INTO migrations VALUES(78,'2024_03_22_000000_update_forum_topic_table',1);
+INSERT INTO migrations VALUES(79,'2024_03_22_00000_update_set_requests_table',1);
+INSERT INTO migrations VALUES(80,'2024_03_23_000000_update_setclaim_table',1);
+INSERT INTO migrations VALUES(81,'2024_03_26_000000_update_news_table',1);
+INSERT INTO migrations VALUES(82,'2024_03_29_000000_update_forum_topic_comment_table',1);
+INSERT INTO migrations VALUES(83,'2024_03_30_000000_update_leaderboarddef_table',1);
+INSERT INTO migrations VALUES(84,'2024_03_30_000001_update_messages_table',1);
+INSERT INTO migrations VALUES(85,'2024_04_03_000000_update_player_sessions',1);
+INSERT INTO migrations VALUES(86,'2024_04_06_000000_update_forum_topic_table',1);
+INSERT INTO migrations VALUES(87,'2024_04_13_000000_update_gamedata_table',1);
+INSERT INTO migrations VALUES(88,'2024_04_27_000000_update_game_sets_table',1);
+INSERT INTO migrations VALUES(89,'2024_05_10_000000_drop_leaderboardentry_table',1);
+INSERT INTO migrations VALUES(90,'2024_05_12_000000_create_achievement_set_game_hashes_table',1);
+INSERT INTO migrations VALUES(91,'2024_05_19_000000_update_achievements_table',1);
+INSERT INTO migrations VALUES(92,'2024_05_19_000000_update_friends_table',1);
+INSERT INTO migrations VALUES(93,'2024_05_19_000000_update_game_hashes_table',1);
+INSERT INTO migrations VALUES(94,'2024_05_20_000000_update_ticket_table',1);
+INSERT INTO migrations VALUES(95,'2024_05_21_000000_update_forumtopiccomment_table',1);
+INSERT INTO migrations VALUES(96,'2024_05_21_000000_update_leaderboarddef_table',1);
+INSERT INTO migrations VALUES(97,'2024_05_21_000000_update_setclaim_table',1);
+INSERT INTO migrations VALUES(98,'2024_05_21_000000_update_siteawards_table',1);
+INSERT INTO migrations VALUES(99,'2024_05_25_000001_update_player_games_table',1);
+INSERT INTO migrations VALUES(100,'2024_06_08_000000_drop_websockets_statistics_entries_table',1);
+INSERT INTO migrations VALUES(101,'2024_06_18_000000_update_gamedata_table',1);
+INSERT INTO migrations VALUES(102,'2024_07_03_000000_update_codenotes_table',1);
```

Comment:
Is there any benefit to inserting these migrations? The sqllite DB shouldn't ever be upgrade or downgraded.

I'm also curious about their necessity for mysql. If these files no longer exist, why track them?

From reading the documentation, it seems like the squash just fast-forwards over the migrations instead of reapplying them all. I don't believe it's intended that the migrations themselves be deleted from the repository. What if someone still has a 6-month old DB that needs migrated?

Reply:
Both of these SQL files are auto-generated by Laravel itself using `php artisan schema:dump`.

>I don't believe it's intended that the migrations themselves be deleted from the repository. What if someone still has a 6-month old DB that needs migrated?

The challenge with this is Laravel 11's breaking change to migrations  ([see here](https://laravel.com/docs/11.x/upgrade#modifying-columns)):
> If you do not want to update all of the existing "change" migrations in your application to retain the column's existing attributes, you may simply [squash your migrations](https://laravel.com/docs/11.x/migrations#squashing-migrations).

My interpretation of this is after we migrate to Laravel 11, the functionality of those migrations isn't going to be the same anyway, and we're better off squashing them all to avoid issues. If we don't do this, the logic of the migrations themselves could potentially change as a result of this breaking change.

If someone has an unmigrated database from 6 months ago, they have a lot more to worry about beyond just migrations. We've remediated tons of data integrity issues over the last 6 months - that person would be better off starting from a fresh DB, in my opinion.

Reply:
> Now, when you attempt to migrate your database and no other migrations have been executed, **Laravel will first execute the SQL statements in the schema file of the database connection you are using**. After executing the schema file's SQL statements, **Laravel will execute any remaining migrations that were not part of the schema dump**.

The way I read it, the schema dump serves as a pre-migrated database to which new migrations will be applied. I don't see anything in the documentation that says to delete the old migrations.

I think I understand the problem though - the migrations may not behave correctly because of the breaking change, so applying them may not work as expected. In that case, I guess it's reasonable to destroy the old files. I'm still uncertain whether there's any benefit in keeping the old migration trail though. Having them there implies the user could still do a `sail artisan migrate:rollback --steps=102`. Removing them from my local file shaved another 0.75 seconds off the parallel run time (though for a sample size of one run that might just be noise).

After we're on Laravel 11, it would make sense to periodically update the schema dumps to avoid running any _new_ migrations added, but those new migrations should not be deleted.

Reply:
>I think I understand the problem though - the migrations may not behave correctly because of the breaking change, so applying them may not work as expected.

Yes, this is the angle I'm seeing things from. I'm nervous keeping the old migrations around could be harmful because they no longer behave as they did when they were written, so even as a historical archive they are inaccurate.

Rather than destroying the migrations, perhaps as a happy medium they could be moved to a non-executed "old", "archive", or "legacy" folder?

Reply:
Technically, people should be using the seeded database, rather than a dump, and I've confirmed that that is working with the raw schema files, so I think the impact of destroying the legacy files is minimal. Users could always go dig them up from github if they need them (even do so conveniently from the PR). They could also just checkout a commit prior to this PR and run the migration directly.

I think removing the files is the correct solution to ensure they don't cause unintended behavior going forward, but we should also remove the references to the files from the imports (both for performance and data integrity). Just delete the last 102 INSERTs from both files, and call it good.

Any migrations added after this PR (even if they're merged first) should remain to ensure the changes get applied to the captured schema files. And they should remain even if the schema files get updated in a future migration squash, assuming they're laravel 11 equivalent (which should be a requirement for future merges even if we haven't made the laravel 11 switch).

For production, I think it's reasonable to keep the legacy migration records even if they can't be rolled back. It's not like we'd do any rollbacks in production anyway.

Reply:
In latest, the 102 INSERTs have been removed from both files. I've also stripped the Pest dependencies.